Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+* Convenience methods - while not necessarily needed, they are nice to have.
+  * `Get<MyCustomType>DisplayNode()` function that returns the downcast version of [`GetDisplayNode()`](http://apidocs.slicer.org/master/classvtkMRMLDisplayableNode.html#ab2cac46d934a10817b75b37471fe65a8) saves users of your class a bit of downcasting.
+* Other methods:
+  * Add other methods as your heart desires to view/modify the actual content of the data being stored.
+
+:::{tip}
+
+Any methods with signatures that contain only primitives, raw pointers to VTK derived objects, or a few std library items like `std::vector` will be automatically wrapped for use in Python. Any functions signatures that contain other classes (custom classes, smart pointers from the std library, etc) will not be wrapped. For best results, try to use existing VTK data objects, or have your custom classes derive from [`vtkObject`](https://vtk.org/doc/nightly/html/classvtkObject.html) to get automatic wrapping.
+
+:::
+
+#### The display node
+
+The display node, contrary to what one may think, is not actually how a MRML object is displayed on screen. Instead it is the list of options for displaying a MRML object. Things like colors, visibility, opacity; all of these can be found in the display node.
+
+Files:
+
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>DisplayNode.h
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>DisplayNode.cxx
+
+Key Points:
+
+* Naming convention for class: `vtkMRML<MyCustomType>DisplayNode`
+  * E.g. [`vtkMRMLModelDisplayNode`](http://apidocs.slicer.org/master/classvtkMRMLModelDisplayNode.html)
+* Inherits from [`vtkMRMLDisplayNode`](http://apidocs.slicer.org/master/classvtkMRMLDisplayNode.html).
+* Constructing a new node is same as the [data node](#the-data-node).
+* To save to an MRB is same as for the [data node](#the-data-node):
+  * Some MRML types like Markups store display information when the actual data is being stored via the writer/storage node. If you do that, no action is needed in this class.
+* Convenience methods:
+  * `Get<YourDataType>Node()` function that returns a downcasted version of [`vtkMRMLDisplayableNode* GetDisplayableNode()`](http://apidocs.slicer.org/master/classvtkMRMLDisplayNode.html#a7100d9379c0b96b2a9ab1ad0a8514af7).
+* Other methods:
+  * Add any methods regarding coloring/sizing/displaying your data node.
+
+#### The widget
+
+The widget is interaction half of actually putting a usable object in the 2D or 3D viewer. It is in charge of making the widget representation and interacting with it.
+
+:::{note}
+
+If your MRML node is display only without any interaction from the viewers, the widget is not necessary, just the [widget representation](#the-widget-representation) for displaying.
+
+:::
+
+Files:
+
+* \<Extension>/\<Module>/VTKWidgets/vtkSlicer\<MyCustomType>Widget.h
+* \<Extension>/\<Module>/VTKWidgets/vtkSlicer\<MyCustomType>Widget.cxx
```

Comment:
```
|-- <Extension>
       |-- <Module>
              |-- VTKWidgets
                    |-- vtkSlicer<MyCustomType>Widget.h
                    |-- vtkSlicer<MyCustomType>Widget.cxx
```

Reply:
Updated to suggested file structure formatting.